awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -88,7 +113,21 @@ def get_all_endpoint(self, endpoint, *args, **kwargs):
             response['json']['next'] = next_page
         return response
 
-    def get_one(self, endpoint, *args, **kwargs):
+    def get_one(self, endpoint, name_or_id=None, *args, **kwargs):
+        if name_or_id:
+            name_field = self.get_name_field_from_endpoint(endpoint)
+            new_args = kwargs.get('data', {}).copy()
```

Comment:
This will result in queries like `/api/v2/inventories/?or__id=42&or__name=42&organization=1`

I'm still reading through the rest of this, but I want to be extra sure that this gets tested, because there is some fundamental ambiguity to this. Is this expressing `(name=42 OR id=42) AND organization=1`, or is it expressing `id=42 OR (name=42 AND organization=1)`? These are very much not the same thing, and failing test cases can be produced depending on your assumed expectations. I feel like it would have been easier to juggle the corner cases that arise from this if we kept the id check as a separate request.

Reply:
just to follow through what this does:

https://github.com/ansible/awx/blob/90c3bfc6ae92aa087f317e7515aeadef5cbfe8f5/awx/api/filters.py#L303-L305

(this code is not easy to read)

https://github.com/ansible/awx/blob/90c3bfc6ae92aa087f317e7515aeadef5cbfe8f5/awx/api/filters.py#L322-L323

https://github.com/ansible/awx/blob/90c3bfc6ae92aa087f317e7515aeadef5cbfe8f5/awx/api/filters.py#L344-L350

If that wasn't hard enough to read, let's look at where the real action happens:

https://github.com/ansible/awx/blob/90c3bfc6ae92aa087f317e7515aeadef5cbfe8f5/awx/api/filters.py#L369

This constructs a crazy complicated queryset in our cases that would (psuedo-code) look like:

```python
Inventory.objects.filter(
  organization=1,
  (Q(name=42) | Q(id=42))
)
```